Chargebee/Classes/Purchase/CBPurchaseManager.swift

```diff
@@ -13,17 +13,19 @@ public class CBPurchase: NSObject {
     private var productIDs: [String] = []
     public var receiveProductsHandler: ((_ result: Result<[CBProduct], CBPurchaseError>) -> Void)?
     public var buyProductHandler: ((Result<(status:Bool, subscriptionId:String?, planId:String?), Error>) -> Void)?
-    
+    public var buyNonSubscriptionProductHandler: ((Result<(NonSubscription), Error>) -> Void)?
+
     private var authenticationManager = CBAuthenticationManager()
     var productRequest: SKProductsRequestFactory = SKProductsRequestFactory()
 
     var restoredPurchasesCount = 0
     private var activeProduct: SKProduct?
     var customer: CBCustomer?
     
-     var restoreResponseHandler: ((Result<[InAppSubscription], RestoreError>) -> Void)?
-     var refreshHandler: RestoreResultCompletion<String>?
-     var includeInActiveProducts = false
+    var restoreResponseHandler: ((Result<[InAppSubscription], RestoreError>) -> Void)?
+    var refreshHandler: RestoreResultCompletion<String>?
+    var includeInActiveProducts = false
+    var productType: ProductType?
```

Comment:
Minor: Since this is only used inside this class, can we pass it as a method arguments instead of storing it at the instance level?

Reply:
Here we are taking productType in ` func purchaseNonSubscriptionProduct(product: CBProduct, customer : CBCustomer? = nil ,productType : ProductType? = nil,completion handler: @escaping ((_ result: Result<NonSubscription, Error>) -> Void))` but we are using it while preparing receipt once the Storekit delegate method have been responded , so only by storing we can use `productType` while validating `receipt` i think.